text/1193207/itai_rodeh.tex

```diff
@@ -0,0 +1,111 @@
+\documentclass{article}
+\usepackage[utf8]{inputenc}
+\usepackage[margin=2cm]{geometry}
+\usepackage{amsthm}
+\newtheorem{lemma}{Lemma}
+
+\usepackage[colorlinks=false]{hyperref}
+\usepackage[backend=bibtex,style=chem-angew,citestyle=numeric-comp,articletitle=true,hyperref=true,url=true]{biblatex}
+\addbibresource{biblio.bib}
+
+\begin{filecontents}{biblio.bib}
+@article{ItaiRodeh1990,
+  doi = {10.1016/0890-5401(90)90004-2},
+  url = {https://doi.org/10.1016/0890-5401(90)90004-2},
+  year = {1990},
+  month = sep,
+  publisher = {Elsevier {BV}},
+  volume = {88},
+  number = {1},
+  pages = {60--87},
+  author = {Alon Itai and Michael Rodeh},
+  title = {Symmetry breaking in distributed networks},
+  journal = {Information and Computation}
+}
+
+\end{filecontents}
+
+
+\begin{document}
+
+\section{Introduction}
+Given a single-directional ring of n processors, it is required to program the processors such that they are able to find the size of the ring (let that size be n). When processors are distinguishable, it is sufficient to count distinct identifiers. Thus, we will consider indistinguishable processors. Itai and Rodeh (1990) \cite{ItaiRodeh1990} have proven that there exists no deterministic, always correct solution to this problem fit for any ring size - mostly due to it being impossible to recognise messages originating from same processor. It is therefore necessary to consider probabilistic approach - an algorithm that always terminates with usually correct result. We shall consider processors with access to random bit generator. Itai and Rodeh (1990) have also proven that there exists no processor terminating (meaning all processors terminated) algorithm that solves this problem which is correct with probability a > 0. Thus, it is necessary to consider message terminating (meaning no processor will send a message without receiving one, and all messages have been received) algorithms. Itai and Rodeh (1990) also propose such an algorithm, which is correct with externally chosen probability 1 > a > 0.
+
+\section{Algorithm}
+\subsection{Overview}
+Throughout the algorithm, each processor maintains it's best estimation of the ring size ($k_n$). This estimation is initially 2 and is guaranteed to never exceed true ring size. In response to a received message, a processor may increase its estimation, or it may increase confidence in its estimation. The minimum confidence is the external parameter that decides probability of correctness. More specifically, let the parameter be $r$; in that case, we define confidence as $r*k_n$ (so dependant on current ring size estimation). The algorithm terminates when all processors are confident in their value of $k_n$. At such a point, no processor initiates a message. It will be shown that when this happens, $k_n$ is same for all processors.
```

Comment:
1. If you want to emphasize multiplication, use $r \cdot k_n$
2. dependant -> dependent (all in all, please run some auto language checker as a sanity check)